### ISSUE TITLE
Delete shortcut on mac

### ISSUE THREAD

**Issue by simonv3** [jdittrich/quickMockup#6]
Maybe watch for a tap of 'delete' or add an 'x' in the top right corner of the element?

**Comment by simonv3** [comment]
Heh, just read the documentation. Already implemented :P

**Comment by simonv3** [comment]
Oh wait. Is this the "del" key on a Windows keyboard? This doesn't work with a Mac's delete key (which is also the backspace key).

**Comment by jdittrich** [comment]
Ah, interesting. Don't have a mac around at the moment. Do Mac users expect objects to delete on backspace? If yes, we could implement this – (FYI this is where the event [binding magic happens](https://github.com/jdittrich/quickMockup/blob/4398d3fc533105ee3cc45b42e9ebe86dcc2c195b/index.html#L941) and this is  [mousetrap](https://craig.is/killing/mice), the library I used, just in case you want to give it a go)

**Comment by simonv3** [comment]
the backspace button is called "delete" so I assume so :P. 

I'll give it a go when I come back from dinner (or maybe tomorrow, but I definitely won't mind digging into the code for this).

**Comment by jdittrich** [comment]
TLDR; On (my) mac (/firefox) the backspace is also mapped to the browser's back function. 

**What should happen**: delete the element
**What happens instead**: deletes the element and (wants to go back and) asks if I want to leave the page.

**Solutions:** 
- prevent the default action (may make sense for these in general, also a windows user is unlikely to want to navigate away by backspace (if that key binding is existing there?)
- use other key

**Comment by simonv3** [comment]
Out of curiousity, is this a binding you added yourself? My Firefox doesn't do that.

**Comment by jdittrich** [comment]
Hmm, did not change anything as far as I remember. Are there other Mac-People around who might enlighten us? (And does that keybinding exist in other browsers too?)

**Comment by paulproteus** [comment]
For what it's worth, Backspace as "back" is a common browser shortcut. See
this page:
https://ux.stackexchange.com/questions/53176/why-does-backspace-go-back-a-page-this-behavior-is-so-frustrating

**Comment by jdittrich** [comment]
@paulproteus: Thanks! Thats a great thing to know. 

I'm unsure of how to proceed – we could:
- Set up an alternative, unambigous delete key
- Your ideas...

**Comment by simonv3** [comment]
Hmm, maybe I did a mapping? 

Maybe just "d"? 

Or we overwrite the behavior. Since it's a webapp that mimics drawing and design apps like Photoshop, Sketch, and Illustrator, I don't think using the "delete/back" key for this purpose is wrong. The question I guess is - are users expecting their browser to act as browsers, or are they expecting standard app functionality? 

From that SE:

```
var hasfocus = 'false';

// when focus happens, set a variable
$(document).on('focusin','input, textarea',function() {
    hasfocus = 'true';
    });

// unset when focus is not happening    
$(document).on('focusout','input, textarea',function() {
    hasfocus = 'false';
    });

// if not in a form field, stop backspace and delete default action
$(document).keypress(function (e) {
    if(e.which == 8 || e.which == 46) {
        if(hasfocus == 'false') {
            e.preventDefault();
           }
       }
    });
```

**Comment by jdittrich** [comment]
> are users expecting their browser to act as browsers, or are they expecting standard app functionality?

Good question. 

I just tried with yahoo mail, they use it as a delete key. 

Though I agree with the stackexcange person saying that it is usually not a great idea to block the keys,  we already break the behaviour with the "are you sure you want to leave" message, so remapping would not do too much additional harm but some additional good here.
- [ ] See some other webapps
- [x] Is the above mentiones block-shortcut function possibly already part of the library we use, [moustrap](https://craig.is/killing/mice)? _Yes,seemingly it is. See the linked page, Headline: Stopping the default behavior_

**Comment by paulproteus** [comment]
I'm +0 to "Replace browser's backspace behavior with your preferred
behavior" in this case.

I'm just a bystander here, really, so I can't vote +1 because that implies
I would actually do something about it, but there you go. (-:

**Comment by simonv3** [comment]
@asheesh, I think you need to inform the masses on GitHub that +1 implies you'll do something about it :P

**Comment by jdittrich** [comment]
check the most current version (the one on the demopage e.g.)  and see if it works (it does for me).

**Comment by simonv3** [comment]
Yep, that works for me!

Edit: Confirmed, both Chrome and Firefox

**Comment by jdittrich** [comment]
I close this optimistically. We can still reopen if virtual stuff explodes :)
